packages/garbo/src/tasks/cockroachPrep.ts

```diff
@@ -16,35 +15,91 @@ import {
 import {
   abort,
   adv1,
+  Effect,
   mallPrice,
   myBuffedstat,
+  myEffects,
+  numericModifier,
+  retrieveItem,
   runChoice,
+  toEffect,
+  use,
   visitUrl,
 } from "kolmafia";
 import { garboValue } from "../garboValue";
 import { freeFightFamiliar } from "../familiar";
-import { freeFightOutfit } from "../outfit";
+import { freeFightOutfit, meatTargetOutfit } from "../outfit";
 import { GarboStrategy, Macro } from "../combat";
 import { acquire } from "../acquire";
 
 // Just checking for the gummi effects for now, maybe can check other stuff later?
 function checkAndFixOvercapStats(): void {
-  if (myBuffedstat($stat`Muscle`) >= 100) {
-    if (have($effect`Gummiheart`)) uneffect($effect`Gummiheart`);
-  }
-  if (myBuffedstat($stat`Mysticality`) >= 100) {
-    if (have($effect`Gummibrain`)) uneffect($effect`Gummibrain`);
-  }
-  if (myBuffedstat($stat`Moxie`) >= 100) {
-    if (have($effect`Gummiskin`)) uneffect($effect`Gummiskin`);
-  }
-  if (
-    myBuffedstat($stat`Moxie`) >= 100 ||
-    myBuffedstat($stat`Mysticality`) >= 100 ||
-    myBuffedstat($stat`Muscle`) >= 100
-  ) {
-    uneffect($effect`Having a Ball!`);
-  }
+  const stats = [$stat`Muscle`, $stat`Moxie`, $stat`Mysticality`];
+  const effects: Effect[] = Object.keys(myEffects()).map((effectName) =>
+    toEffect(effectName),
+  );
+
+  stats.forEach((stat) => {
+    while (myBuffedstat(stat) > 100) {
+      if (
+        !have($effect`Mush-Mouth`) &&
+        mallPrice($item`Fun-Guy spore`) < 5_000
```

Comment:
don't use an arbitrary price cap, use a non-arbitrary price cap

Reply:
I'm not sure the relative value of each of these or even how to determine that, given all they do for us is... reduce stats.

Reply:
Marking as resolved because we just choose the cheapest thing, now.